src/App.jsx

```diff
@@ -1,14 +1,34 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messages from './data/messages.json';
 
 const App = () => {
+  const [counter, setCounter] = useState(0); // Initial counter set to 0
+
+  const onLike = () => {
+    setCounter((prevCounter) => prevCounter + 1);
+  };
+
+  const onUnlike = () => {
+    setCounter((prevCounter) => prevCounter - 1);
+  };
```

Comment:
Instead of two methods for handling liking and unliking a ChatEntry, you can write an event handler that handles both scenarios. 

That would look something like having an event handler that changes the state of the `ChatEntry`s from `App` and then using another helper function that looks at the state of all the ChatEntries to calculate the heart counts:
```js
  const handleLikeData = (id) => {
    const updatedEntries = chatMessages.map(chatMessage => {
      if (chatMessage.id === id) {
        return { ...chatMessage, liked: !chatMessage.liked };
      }
      else {
        return chatMessage;
      }
    })
    setChatMessages(updatedEntries);
  };

  const calculateTotalLikes = () => 
    return chatMessages.reduce((totalLikes, chatMessage) => (totalLikes + chatMessage.liked), 0);
  };
```

You can see how we do this with pets on a Cat in flasky [here](https://github.com/Ada-C22/flasky-front-end/blob/db8e1afd6d2e4cadf627841adcb1baec55544857/src/App.jsx#L44-L55).